Detectors/FIT/FT0/simulation/src/Detector.cxx

```diff
@@ -358,6 +361,398 @@ void Detector::SetOneMCP(TGeoVolume* ins)
   //  ins->AddNode(alsup,1);
 }
 
+TGeoVolume* Detector::ConstructFrameGeometry()
+{
+  // define the media
+  TGeoMedium* Vacuum = gGeoManager->GetMedium("FT0_Vacuum$");
+  TGeoMedium* Al = gGeoManager->GetMedium("FT0_Aluminium$");
+
+  // make a volume assembly for the frame
+  TGeoVolumeAssembly* FT0_Frame = new TGeoVolumeAssembly("FT0_Frame");
+
+  // define translations for the quartz radiators and PMTs
+  defineTransformations();
+
+  // approximate the frame with some rectangles
+  TGeoBBox* frame1 = new TGeoBBox("frame1", frame1X / 2, frame1Y / 2, frameZ / 2);
+  TGeoBBox* frame2 = new TGeoBBox("frame2", frame2X / 2, frame2Y / 2, frameZ / 2);
+  TGeoBBox* quartzRadiator = new TGeoBBox("quartzRadiator", quartzRadiatorSide / 2, quartzRadiatorSide / 2, quartzRadiatorZ / 2);
+  TGeoBBox* rect1 = new TGeoBBox("rect1", rect1X / 2, rect1Y / 2, frameZ / 2);
```

Comment:
As we discussed, you are indeed using epsilon for some of these, but not for all of them. And all of them are subtracted from something according to lines: 487-494, so at least one dimension should be 'epsilonized'. Unless epsilon is included already in the frame1X, frame1Y constants... Is it?